lib/dangermattic/plugins/labels_checker.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin for checking labels associated with a pull request.
+  #
+  # @example Checking for specific labels and generating warnings/errors:
+  #   labels_checker.check(
+  #     do_not_merge_labels: ['Do Not Merge'],
+  #     required_labels: ['Bug', 'Enhancement'],
+  #     required_labels_error: 'Please ensure the PR has labels "Bug" or "Enhancement".',
+  #     recommended_labels: ['Documentation'],
+  #     recommended_labels_warning: 'Consider adding the "Documentation" label for better tracking.'
+  #   )
+  #
+  # @see Automattic/dangermattic
+  # @tags github, process
+  #
+  class LabelsChecker < Plugin
+    DEFAULT_DO_NOT_MERGE_LABELS = [
+      'Do Not Merge'
+    ].freeze
+
+    # Checks if a PR is missing labels or is marked with labels for not merging.
+    # If labels are missing, the plugin will emit a warning. If a label indicating that the PR should not be merged is present,
+    # an error will be emitted, preventing the final PR merge.
+    #
+    # @param do_not_merge_labels [String] The possible labels indicating that a merge should not be allowed.
+    #   Defaults to DEFAULT_DO_NOT_MERGE_LABELS if not provided.
+    # @param required_labels [Array<RegExp>] The list of Regular Expressions describing all the type of labels that are *required* on PR (e.g. `[/^feature:/, `/^type:/]` or `bug|bugfix-exemption`).
```

Comment:
Ruby's regular expression type is named `Regexp` not `RegExp` (I get tripped by this all the time 😅 )
```suggestion
    # @param required_labels [Array<Regexp>] The list of Regular Expressions describing all the type of labels that are *required* on PR (e.g. `[/^feature:/, `/^type:/]` or `bug|bugfix-exemption`).
```

Reply:
Oof, true.